generator/src/main/java/org/apache/kafka/message/MessageSpecType.java

```diff
@@ -30,5 +30,8 @@
     HEADER,
 
     @JsonProperty("data")
-    DATA;
+    DATA,
+
+    @JsonProperty("metadata")
```

Comment:
Might be helpful to have a short description here (or somewhere else if you can think of a better location) which explains that this type is reserved for the controller record types.

Reply:
I added a comment